src/main/java/core/basesyntax/service/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.model.OperationType;
+import core.basesyntax.model.impl.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.OperationStrategy;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy<FruitTransaction> {
+    private final Map<OperationType, OperationHandler<FruitTransaction>> operationHandlerMap;
+
+    public OperationStrategyImpl(Map<OperationType,
+            OperationHandler<FruitTransaction>> operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+    }
+
+    @Override
+    public OperationHandler<FruitTransaction> get(FruitTransaction transaction) {
+        return operationHandlerMap.get(transaction.getTransactionType());

```

Comment:
I also do not like the idea of passing the whole transaction here, where we only need it's type, that's one of the reasons why if we fix that - it will be just a map with non-direct access, which is kind of strange